<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <!-- Camera parameters -->
  <arg name="rgb_image_topic" default="kinect2/qhd/image_color_rect"/>
  <arg name="rgb_camera_info_topic" default="kinect2/qhd/camera_info"/>
  <arg name="depth_image_topic" default="kinect2/qhd/image_depth_rect"/>
  <arg name="depth_camera_info_topic" default="kinect2/qhd/camera_info"/>

  <arg name="global_frame_id" default="map"/>

  <!-- Tracker parameters -->
  <arg name="n_init" default="3"/>
  <arg name="n_frame" default="4"/>
  <arg name="max_lost" default="6"/>
  <arg name="max_age" default="20"/>
  <arg name="max_iou_distance" default="0.98"/>

  <!-- Other parameters -->
  <arg name="publish_viz" default="true"/>
  <arg name="publish_tf" default="true"/>
  <arg name="use_depth" default="false"/>

  <node name="tabletop_object_perception" pkg="uwds3" type="tabletop_object_perception_node.py" args="" output="screen" respawn="false">
    <rosparam subst_value="true">

      rgb_image_topic: $(arg rgb_image_topic)
      rgb_camera_info_topic: $(arg rgb_camera_info_topic)

      depth_image_topic: $(arg depth_image_topic)
      depth_camera_info_topic: $(arg depth_camera_info_topic)

      global_frame_id: $(arg global_frame_id)

      publish_viz: $(arg publish_viz)
      publish_tf: $(arg publish_tf)

      use_depth: $(arg use_depth)

      n_init: $(arg n_init)
      n_frame: $(arg n_frame)
      max_iou_distance: $(arg max_iou_distance)
      max_lost: $(arg max_lost)
      max_age: $(arg max_age)
    </rosparam>
  </node>

</launch>
